Plugins/Json/include/Acts/Plugins/Json/IndexedGridJsonHelper.hpp

```diff
@@ -0,0 +1,253 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Detector/detail/GridAxisGenerators.hpp"
+#include "Acts/Plugins/Json/AlgebraJsonConverter.hpp"
+#include "Acts/Plugins/Json/GridJsonConverter.hpp"
+#include "Acts/Utilities/detail/AxisFwd.hpp"
+#include "Acts/Utilities/detail/Grid.hpp"
+
+#include <tuple>
+
+namespace Acts {
+using namespace Experimental::detail::GridAxisGenerators;
+
+namespace IndexedGridJsonHelper {
+
+// Generate the possible axes in this case
+static auto s_possibleAxes =
+    std::tuple<EqBound, EqOpen, EqClosed,
```

Comment:
Just to mention that `tuple` is not very efficient with many types, so if at all possible, avoid using a giant tuple.

Reply:
This is for JsonIO, so I think we might be anyway limited by the IO and this is not performance critical.

Reply:
It's inefficient at compile time, since it recursively instantiates structs to generate the type.

Reply:
Oh, I see - so you have another suggestion how to implement this then?

Reply:
So if I read the code right, you're using the tuple only to communicate the types, and not actual values (correct me if I'm wrong). So effectively you're using it to wrap a parameter pack, rather than just using a parameter pack? I don't think I see anywhere the tuple values are actually unpacked. 

And if that's anywhere, what does that even mean: here's an object of ALL axis types?

So I think I'd suggest using the parameter pack directly. If you don't want to list the type more than once (which I don't think is the case because you only call `unrollConvert` in one spot, use a struct/meta-function that encodes only the types.

Reply:
Reading the code again, yeah since you never actually put values in that tuple, you really don't need the tuple at all, I don't think.

Reply:
I think this is normally called a "type list", and you can use it to re-extract the parameter pack wherever you need.

https://www.codingwiththomas.com/blog/getting-started-with-typelists

Reply:
Ok, thanks, I will have a look at that.

Reply:
That's exactly what I used for the detray grid IO :)
https://github.com/acts-project/detray/blob/main/core/include/detray/utils/type_list.hpp

Reply:
With mini - unittest:
https://github.com/acts-project/detray/blob/main/tests/unit_tests/core/utils_type_list.cpp

Reply:
Is this a Utility we could use in Acts/Core in general?

Reply:
I suppose, in the detray implementation would also create this tuple of types:

```c++
template <int N, typename T, typename... Ts>
struct get_at<N, type_list<T, Ts...>> {
    using type = detail::remove_cvref_t<decltype(
        detray::get<N>(detray::tuple<T, Ts...>{}))>;
};
```

This looks to me that the tuple is created at this stage, I will try to see whether I can iterate through without instantiating that tuple.

Reply:
I think I can just do simply with a `<Head, Tail...>` pattern.

Reply:
In the end, I don't believe I actually used that method, but yes, I guess it can be done without a tuple. In detray, the axes are built by deducing the pack in the building function:
```

    /// @brief End of recursion: build the grid from the @param grid_data
    template <typename... bounds_ts,
              typename... binning_ts>
    static void deserialize(types::type_list<bounds_ts...>,
                            types::type_list<binning_ts...>) {
 and then doing:
```
n_axis::single_axis<bounds_ts, binning_ts>...
```

Reply:
TIL you can have multiple parameter packs.

Reply:
If you end up keeping `get_at` with a roundtrip to tuple, you still avoid having tuple on the other code paths.

Reply:
All done.